docs/getting_started/overview.rst

```diff
@@ -0,0 +1,76 @@
+Bird's eye view on ImageIO
+==========================
+
+The aim of this page is to give you a high level overview of how ImageIO works
+under the hood. We think this is useful for two reasons: 
+
+#.  If something doesn't work as it should, you need to know where to search for
+    causes. The overview on this page aims to help you in this regard by giving you
+    an idea of how things work, and - hence - where things may go sideways. 
+#.  If you do find a bug and decide to report it, this page helps us establish
+    some joint vocabulary so that we can quickly get onto the same page, figure out
+    what's broken, and how to fix it.
+
+
+Terminology
+-----------
+
+You can think of ImageIO in three parts that work in sequence in order to load
+an image.
+
+ImageIO Core 
+    The user-facing APIs (legacy + v3) and a plugin manager. You
+    send requests to iio.core and it uses a set of plugins (see below) to figure out
+    which backend (see below) to use to fulfill your request. It does so by directly
+    sending the request to the plugin you specified or by (intelligently) searching
+    all plugins if you didn't specify one explicitly.
+
+Plugin
+    A backend-facing adapter/wrapper that responds to a request from
+    iio.core. It can convert a request that comes from iio.core into a sequence of
+    instructions for the backend that fullfill the request (eg., read/write/iter). A
+    plugin is also aware if its backend is or isn't installed and handles this case
+    appropriately, e.g., it deactivates itself if the backend isn't present.
+
+Backend 
```

Comment:
Idea: maybe we should call this "backend library" to make the distinction with plugin more clear.

Reply:
I like your thinking. Is it always a library? E.g., thinking of FFMPEG, we actually call an executable, but I guess this is wrapped into a library (imageio-ffmpeg / pyav / ...) and further this distinction may not be too important outside of plugin development.

That said, I'm happy to call it backend libraries or backend <something>.

Reply:
> Is it always a library?

I guess it usually is. For the backends included in imageio it's technically a sub-package, but  ... close enough? I sometimes find myself slightly confused between `backend` and `plugin`, and I imagine it will be worse for people other than us. So anything to help reduce that confusion would help :)

Reply:
Yep, I agree. I don't see anything right now that breaks the abstraction of calling it a "library", so let's roll with it and see how it goes.

Reply:
done :)